backend/src/main/java/se/chalmers/dat265/group1/api/order/OrderController.java

```diff
@@ -0,0 +1,138 @@
+            }
+            if (!alreadyFound) {
+                List<OrderedPart> singleList = new LinkedList<>();
+                singleList.add(orderedPart);
+                Order order = new Order(tempData, singleList);
+                orders.add(order);
+            }
+        }
+        return orders;
+    }
+
+    private List<Order> getMerged(List<OrderData> orderDataList) {
+        List<Order> orders = new LinkedList<>();
+        for (OrderData orderData : orderDataList) {
+            Order temp = new Order(orderData, orderedPartRepository.getObjects("orderID=" + orderData.getId()));
+            orders.add(temp);
+        }
+        return orders;
+    }
+
+    @Override
+    public Order getOrder(String orderID) {
+        OrderData orderData = orderDataRepository.getObject(Integer.parseInt(orderID));
+        return new Order(orderData, orderedPartRepository.getObjects("orderID=" + orderData.getId()));
+    }
+
+    @Override
+    public Order createNewOrder(Order order) throws InvalidDateFormatException {
+        validateDate(order.getDate());
+        int id = orderDataRepository.postObject(new OrderData(order)).getId();
+        for (OrderedPart orderedPart : order.getOrderedParts()) {
+            orderedPart.setOrderID(id);
+            orderedPartRepository.postObject(orderedPart);
+        }
+        return getOrder(id + "");
+    }
+
+    @Override
+    public Order updateOrder(String orderID, Order order) throws InvalidDateFormatException {
+        checkIDs(orderID, order);
+        validateDate(order.getDate());
+
+        OrderData orderData = orderDataRepository.updateObject(new OrderData(order));
+
+        List<OrderedPart> orderedInDb = orderedPartRepository.getObjects("orderID=" + orderID);
+        List<OrderedPart> orderedNew = order.getOrderedParts();
+        List<OrderedPart> toPostToDb = new LinkedList<>();
+
+        for (OrderedPart newOrdered : orderedNew) {
+            newOrdered.setOrderID(orderData.getId());
+            boolean preExisting = false;
+            for (OrderedPart dbOrderdPart : orderedInDb) {
+                if (dbOrderdPart.getId() == newOrdered.getId()) {
+                    orderedPartRepository.updateObject(newOrdered);
+                    preExisting = true;
+                    orderedInDb.remove(dbOrderdPart);
+                    break;
+                }
+            }
+            if (!preExisting) {
+                toPostToDb.add(newOrdered);
+            }
+        }
+        for (OrderedPart newOrdered : toPostToDb) {
+            orderedPartRepository.postObject(newOrdered);
+        }
+
+        for (OrderedPart dbOrderdPart : orderedInDb) {
+            orderedPartRepository.deleteObject(dbOrderdPart.getId());
+        }
+        return getOrder(orderID);
+    }
+
+    /**
+     * Dates should be on format yyyy-MM-dd
+     *
+     * @param date
+     */
+    private void validateDate(String date) throws InvalidDateFormatException {
+        //TODO
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)